x/wasm-storage/keeper/filter.go

```diff
@@ -39,3 +54,55 @@ func ApplyFilter(filter []byte, reveals []RevealBody) ([]int, bool, error) {
 		return outliers, false, nil
 	}
 }
+
+// filterMode takes in a list of reveals, and a json path to get the
+// filtering data from the RevealBody->reveal.
+// The function returns a list of int where value of each element can be
+// either 0 or 1. At index i Value 1 means RevealBody[i] contains an
+// outlier. If a reveal have MORE THAN 2/3 frequency, than it's not an
+// outlier.
+//
+// The function also returns if a consensus is reached. Consensus can be
+// reached in two ways.
+// 1. More than 1/3 of the reveals are corrupted. i.e. has non-zero exit code.
+// 2. More than 2/3 of the data is the same.
```

Comment:
My understanding is that consensus is reached as long as more than 2/3 of results, which include both the reveal data and the exit codes, are the same. 
Perhaps we just don't want to distinguish between different non-zero exit codes? 
In any case, I'm confused why we separate the logic for checking for consensus into two: checking the reveal data and checking the exit codes. We can loop through the reveals, counting the frequencies of an occurrence of a given reveal data - exit code pair, and then determine the consensus outcome at once.

Reply:
I see what you mean now. Please ignore the last message.

Reply:
For the input 
A0,B1,B1

If you think the output should be
[1,0,0] true;

Then we have different opinion about consensus.

For this input, result should be
[1,1,1] true

All are outliers as we cannot trust 1/3 of the reveals.
And we are in consensus that the output is garbage.




If you are still confused, let's huddle.